packages/keyring-controller/src/KeyringController.ts

```diff
@@ -2146,6 +2161,19 @@ export class KeyringController extends BaseController<
     return serializedKeyrings;
   }
 
+  /**
+   * Get a snapshot of data that can be updated during a `withKeyring` operation.
+   *
+   * @returns Snapshot of updatable data.
+   */
+  async #getUpdatableState(): Promise<UpdatableState> {
+    return {
+      keyrings: await this.#getSerializedKeyrings(),
+      keyringsMetadata: this.#keyringsMetadata.slice(), // Force copy.
+      password: this.#password,
+    };
+  }
+
```

Comment:
nit: As "state" may be confused with the actual controller state, and as this is used basically everywhere rather than just `withKeyring`, what do you think about changing the wording to something like this?

```suggestion
  /**
   * Get a snapshot of session data held by class variables.
   *
   * @returns An object with serialized keyrings, keyrings metadata,
   * and the user password.
   */
  async #getControllerSessionState(): Promise<SessionState> {
    return {
      keyrings: await this.#getSerializedKeyrings(),
      keyringsMetadata: this.#keyringsMetadata.slice(), // Force copy.
      password: this.#password,
    };
  }
```

Reply:
That does sounds a lot better and well aligned with `withKeyring` semantic IMO (== session)